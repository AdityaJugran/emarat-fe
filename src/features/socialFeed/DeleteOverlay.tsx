import { Button, Dropdown, Menu } from 'antd';
import { HiOutlineDotsVertical } from 'react-icons/hi';

type DeleteOverlayType = {
	itemKey: string;
	handleClick: (itemKey: string) => unknown;
};

export default function DeleteOverlay({
	itemKey,
	handleClick,
}: DeleteOverlayType) {
	const onClick = async () => {
		await handleClick(itemKey);
	};
	return (
		<Dropdown
			trigger={['click']}
			overlay={
				<Menu>
					<Menu.Item key="delete">
						<Button
							className="text-red-500"
							type="link"
							onClick={onClick}
						>
							Delete
						</Button>
					</Menu.Item>
				</Menu>
			}
			placement="bottomRight"
		>
			<div className="cursor-pointer h-fit">
				<HiOutlineDotsVertical className="text-gray-400 text-2xl" />
			</div>
		</Dropdown>
	);
}
