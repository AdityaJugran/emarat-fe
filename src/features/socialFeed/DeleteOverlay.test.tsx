import DeleteOverlay from 'features/socialFeed/DeleteOverlay';
import renderer from 'react-test-renderer';

it('Delete Overlay', () => {
	const tree = renderer
		.create(<DeleteOverlay itemKey={itemKeyMock} handleClick={jest.fn()} />)
		.toJSON();
	expect(tree).toMatchSnapshot();
});

const itemKeyMock = '-MjP6ufjLnHm8rWFc_wx';
