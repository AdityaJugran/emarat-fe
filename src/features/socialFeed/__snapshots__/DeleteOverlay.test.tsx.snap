// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Delete Overlay 1`] = `
<div
  className="ant-dropdown-trigger cursor-pointer h-fit"
  onClick={[Function]}
  onMouseDown={[Function]}
  onTouchStart={[Function]}
>
  <svg
    className="text-gray-400 text-2xl"
    fill="none"
    height="1em"
    stroke="currentColor"
    strokeWidth="0"
    style={
      Object {
        "color": undefined,
      }
    }
    viewBox="0 0 24 24"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M12 5v.01M12 12v.01M12 19v.01M12 6a1 1 0 110-2 1 1 0 010 2zm0 7a1 1 0 110-2 1 1 0 010 2zm0 7a1 1 0 110-2 1 1 0 010 2z"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="2"
    />
  </svg>
</div>
`;
